### PR TITLE
Add a blog post about using a custom query

### PR DESCRIPTION
**Pull request by iobrado**
# Description

Blog post covering the topic of using custom queries in your project and a brief explanation of the service classes.

As always, feedback is much appreciated and if you think anything else should be covered here, let me know.

### CODE REVIEW

File: website/blog/2023-08-03-adding-custom-query.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+title: Adding a custom query
```
Comment (goranalkovic-infinum) [inline website/blog/2023-08-03-adding-custom-query.md:2]:
```suggestion
title: Working with custom queries
```

File: website/blog/2023-08-03-adding-custom-query.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+description: Explains the process of registering a new service class, adding a custom query and using it in a block.
+slug: adding-custom-query
```
Comment (goranalkovic-infinum) [inline website/blog/2023-08-03-adding-custom-query.md:4]:
```suggestion
slug: working-with-custom-queries
```
(if you decide to accept the above one)

File: website/blog/2023-08-03-adding-custom-query.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+authors: obradovic
+
+date: 2023-08-03
+tags: [eightshift, boilerplate, service, class, query]
+hide_table_of_contents: false
+---
+
+Eightshift Dev Kit offers some blocks with query logic out-of-the-box, but what is the best approach when you need to add a custom query to a block you’ve been working on?
```
Comment (goranalkovic-infinum) [inline website/blog/2023-08-03-adding-custom-query.md:12]:
```suggestion
Eightshift DevKit offers some blocks with query logic out of the box, but what is the best approach when you need to add a custom query to a block you’ve been working on?
```

**Review by iruzevic — APPROVED** [review]
awesome

File: website/blog/2023-08-03-adding-custom-query.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+<!--truncate-->
+
+## Already existing blocks
```
Comment (goranalkovic-infinum) [inline website/blog/2023-08-03-adding-custom-query.md:15]:
```suggestion
## Usage in built-in blocks
```

File: website/blog/2023-08-03-adding-custom-query.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+// Get 10 latest posts from the News category. News category ID is 2.
+$latestNews = apply_filters(CustomQuery::GET_POSTS_BY_CATEGORY, 2, 10);
+
+// Get another page of News category.
+$pagedNews = apply_filters(CustomQuery::GET_POSTS_BY_CATEGORY, 2, 10, $currentPage);
+```
+
+The above example shows multiple use cases. The first two examples could be used in a simple block that displays only the selected number of the latest posts. The last example may be used in a REST route for a load more functionality or in a block with classic pagination.
+
+Now you can do a regular query loop in your block to display the posts:
```
Comment (goranalkovic-infinum) [inline website/blog/2023-08-03-adding-custom-query.md:95]: Even though it's there, maybe we should mention that `wp_reset_postdata()` shouldn't be forgotten.

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Left a couple of suggestions, awesome job!

File: website/blog/2023-08-03-adding-custom-query.md (hunk 1 above)
Comment (iobrado) [reply website/blog/2023-08-03-adding-custom-query.md:2]: Yes, absolutely! 😄 Coming up with a good title on this one was killing me 😅

File: website/blog/2023-08-03-adding-custom-query.md (hunk 5 above)
Comment (iobrado) [reply website/blog/2023-08-03-adding-custom-query.md:95]: Yep, I agree. I'll update it 🙂

**Review by dadadavorin — DISMISSED** [review]

File: website/blog/2023-08-03-adding-custom-query.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+One of the Eightshift blocks that already uses `WP_Query` is the ***Featured Content*** block. In that block, you can see the query logic is inside the block. However, there is a much better way to do it. The reason it was done like this was to simplify this block and to have an already functioning block available with one WP-CLI command.
+
+A much better approach would be to separate the query logic from the block. Other than following the MVC architecture more closely, this will also make the query logic more reusable. To do this, we’re gonna create a service class.
+
+## What are Service classes?
+
+*Put simply, a Service is any PHP object that performs some sort of "global" task. - Symfony docs*
+
+If you take a look at the `ServiceInterface` interface, you’ll notice it only  contains the `register()` method. This method holds action and filter hooks for that class. Other than hooking into existing actions and filters, this method can be used to register our own filters which can be used in blocks or other classes.
```
Comment (piqusy) [inline website/blog/2023-08-03-adding-custom-query.md:25]:
```suggestion
If you take a look at the `ServiceInterface` interface, you’ll notice it only contains the `register()` method. This method holds action and filter hooks for that class. Other than hooking into existing actions and filters, this method can be used to register our own filters which can be used in blocks or other classes.
```

File: website/blog/2023-08-03-adding-custom-query.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+Basically, whenever you need to hook into actions or filters, you use a Service class for that.
```
Comment (piqusy) [inline website/blog/2023-08-03-adding-custom-query.md:27]:
```suggestion
Basically, whenever you need to hook into actions or filters, you should use a Service class for that.
```

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 
Left two suggestions

**Review by piqusy — APPROVED** [review]

**Review by goranalkovic-infinum — APPROVED** [review]

**Pull request merged**